Sources/BraintreePayPalNativeCheckout/BTPayPalNativeRequest.swift

```diff
@@ -105,58 +105,59 @@ import BraintreePayPal
         switch paymentType {
         case .checkout:
             guard let request = request as? BTPayPalNativeCheckoutRequest else { return [:] }
+            var checkoutParameters: [String: Any] = [
+                "intent": request.intentAsString,
+                "amount": request.amount,
+                "offer_pay_later": request.offerPayLater
+            ]
 
-            var billingAgreementDictionary: [AnyHashable: Any]? = [:]
+            let currencyCode = request.currencyCode ?? configuration.json["paypal"]["currencyIsoCode"].asString()
 
-            if request.billingAgreementDescription != nil {
-                billingAgreementDictionary?["description"] = request.billingAgreementDescription
-            } else {
-                billingAgreementDictionary = nil
+            if currencyCode != nil {
+                checkoutParameters["currency_iso_code"] = currencyCode
             }
```

Comment:
Take it or a leave it comment to use an if-let combining with the line above:

```suggestion
            if let currencyCode = request.currencyCode ?? configuration.json["paypal"]["currencyIsoCode"].asString() {
                checkoutParameters["currency_iso_code"] = currencyCode
            }
```